sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingBufferedSender.cs

```diff
@@ -78,29 +78,24 @@ public class SearchIndexingBufferedSender<T> : IDisposable, IAsyncDisposable
         private Task _getKeyFieldAccessorTask;
 
         /// <summary>
-        /// Async event raised whenever an indexing action is added to the
-        /// sender.
+        /// Event raised whenever an indexing action is added to the sender.
         /// </summary>
-        public event Func<IndexDocumentsAction<T>, CancellationToken, Task> ActionAddedAsync;
+        public event SyncAsyncEventHandler<IndexActionEventArgs<T>> ActionAdded;
```

Comment:
Does having this as non-virtual work OK with the client instrumentation?

Reply:
For Search, yes, because the handler lives on an uninstrumented type (which delegates all of its calls to an instrumented `SearchClient` under the hood).  I think it might need to be `virtual` if it's on a `Client`.

Reply:
That's what I worry about. I understand FDG do not recommend virtual events.

Reply:
I would say that's no different than the FXDG not recommending the approach we take for mocking by making all methods virtual, unless there are other technical issues to worry about.  In the absolute worst case, we can add `internal virtual void AddFooHandler(...); internal virtual void RemoveFooHandler(...);` methods to work around this without virtual events.

Reply:
So would our guideline be to make all events virtual?